### PR TITLE
feat: add is anyof operator

### PR DESCRIPTION
**Pull request by arththebird**
## Description

Add `is anyof` operators as an alternative syntax to `{'foo' 'bar'} contains some_literal`. Rename `ContainsOperator` to `SetComparisonOperator` for better clarity.

## Related changes

Will be merged after
* #2  

## Checklist

- [X] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines 
- [X] All relevant PR sections are populated, irrelevant ones are removed 

## Validation
- [X] All tests pass

image

### CODE REVIEW

**Comment by arththebird** [comment]
>Note that since we're actually using the contains operator under the hood, predicate description and failed evaluation messages will refer to the ContainsOperator.

This was the easiest way to implement it, let me know if you think we should create another distinct operator for this case to have more accurate predicate description and failed evaluation messages.

**Comment by cryptoshell** [comment]
> > Note that since we're actually using the contains operator under the hood, predicate description and failed evaluation messages will refer to the ContainsOperator.
> 
> This was the easiest way to implement it, let me know if you think we should create another distinct operator for this case to have more accurate predicate description and failed evaluation messages.

I think for someone newly contributing, this might be confusing to them since they'll lack the background context.  @arththebird would adding a different description and warning for `is anyof` bring in a lot of complexity?

**Comment by edouardlp** [comment]
> > > Note that since we're actually using the contains operator under the hood, predicate description and failed evaluation messages will refer to the ContainsOperator.
> > 
> > 
> > This was the easiest way to implement it, let me know if you think we should create another distinct operator for this case to have more accurate predicate description and failed evaluation messages.
> 
> I think for someone newly contributing, this might be confusing to them since they'll lack the background context. @arththebird would adding a different description and warning for `is anyof` bring in a lot of complexity?

The problem is perhaps that we should rename `ContainsOperator` to be more generic, something like `SetComparisonOperator` or something along those lines. It makes intuitive sense that `is anyof` and `contains` have a similar representation, it just the current naming that is confusing IMO.

File: pypred/parser.py [hunk 1]
```diff
@@ -185,6 +185,12 @@ def p_contains_all(p):
     p[0] = ast.ContainsOperator("all", p[1], p[4])
     p[0].set_position(p.lineno(2), compute_column(p.lexer, p.lexpos(2)))
 
+def p_term_is_any(p):
+    "term : factor IS_EQUALS ANY factor"
+    # We use the contains operator and reverse the left and right side
+    p[0] = ast.ContainsOperator("contains", p[4], p[1])
```
Comment (edouardlp) [inline pypred/parser.py]: How much of an effort is it to introduce a new `type` that inverts these at comparison time? I fear that once again the AST representation could be a bit confusing to look at.

Comment (arththebird) [reply pypred/parser.py]: I'm in the process of refactoring it to be `SetComparisonOperator` with type `is_any`, should be completed soon :)

Comment (edouardlp) [reply pypred/parser.py]: Nice!

**Review by edouardlp — APPROVED** [review]
👏 🏆 is anyof 💰🍾🏝?

**Pull request merged**